app/controllers/documents_controller.rb

```diff
@@ -27,7 +27,21 @@ def create
         begin
           @document.doc.attach(params[:document][:doc])
         rescue ActiveRecord::RecordNotSaved => e
-          @errors << "Failed to save attachment"
+          @errors << 'Failed to save attachment'
+          raise ActiveRecord::Rollback
+        end
+
+        begin
+          @permission = Permission.create(user_id: current_user.id, document_id: @document.id, ability: 'READ')
```

Comment:
Permission.create will not raise an error if saving fails (it will just return an empty Permission). 

Use `create!` instead of `create`. Also, we can move this into the same block as

```ruby
@document = Document.create!(params.require(:document).permit(:title))
```

and just use the same (existing) error handling there (i.e. `@errors << e.message`)